andela/Ah-backend-xmen
# #162725277 Enable users to see their reading stats

#### What does this PR do?
This PR enables a user to be able to see their reading statistics

#### Description of Task to be completed?
- create a read_stats field that will handle the data from the calculated number of times the article has been read.
- create a view file that will handle the endpoint that retrieves the calculated statistics.
- add logic for when the article is read

#### How should this be manually tested?
- Fetch this branch by running the command `git fetch origin ft-reading-stats-162725277` in the terminal and check into the branch `git checkout ft-reading-stats-162725277`
- Start the virtual environment `source venv/bin/activate`
- Migrate to add the read_stats field to the database, the field required for the data are initiated by typing this command: `python manage.py migrate`
- Start the application by running this command: `python manage.py runserver`
- create an article an then log out and login in as a different user and then read that aricle, you will see that the statistics change

#### Screenshots
Reading the aricle `/api/articles/<article-slug>`
<img width="1360" alt="screen shot 2019-01-30 at 2 32 50 pm" src="https://user-images.githubusercontent.com/34310682/51978743-1bb73380-249c-11e9-9231-cc08993d4c0b.png">

Getting read statistics for a user `/api/profiles/<username>/readstats`
<img width="1358" alt="screen shot 2019-01-30 at 2 32 20 pm" src="https://user-images.githubusercontent.com/34310682/51978883-9718e500-249c-11e9-88d9-fc8c1325226e.png">

[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/e692e9d1ae9e6885a02e)

#### Related story
[#162725277](https://www.pivotaltracker.com/story/show/162725277)
